proposed/2022/DotnetUpdatePackage.md

```diff
@@ -0,0 +1,179 @@
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+This command will update dependencies to the latest version respecting the semver constraints of your package and its transitive dependencies. It will also install any missing packages if an newer version brings them in. It will take into account lock files and central package management as locations that exist in addition to common definitions of `<PackageReference>` in project file(s).
+
+To ensure compatibility with existing package definitions in various files, we should not overwrite or simplify XML elements, but rather only update package version attributes.
+
+![](../../meta/resources/dotnetupdatepackage/dotnetupdatepackage.png)
+
+```
+dotnet update [<PROJECT>|<SOLUTION>] package <PACKAGE_NAME>
+    [-f|--framework <FRAMEWORK>] [--interactive]
+    [-n|--no-restore] [--package-directory <PACKAGE_DIRECTORY>]
+    [--prerelease] [-s|--source <SOURCE>] [-v|--version <VERSION>]
+    [--dry-run] [--highest-minor] [--highest-patch]
+
+dotnet update package -h|--help
+```
+
+#### Arguments
+
+- PROJECT | SOLUTION
+
+The project or solution file to operate on. If not specified, the command searches the current directory for one. If more than one solution or project is found, an error is thrown.
+
+- PACKAGE_NAME
+
+The package reference to add. Package name wildcards should be supported and update a subset of a glob-style pattern package name. i.e. `Microsoft.*` would update Microsoft packages. The commonly used symbols to support are:
+
+- `*` - zero or more characters.
+- `?` - single occurrence of any character.
+- `.` - literal "." character
+
+#### Options
+
+- -f|--framework <FRAMEWORK>
+
+Adds a package reference only when targeting a specific framework.
+
+- -?|-h|--help
+
+Prints out a description of how to use the command.
+
+- --interactive
+
+Allows the command to stop and wait for user input or action. For example, to complete authentication.
+
+- -n|--no-restore
+
+Adds a package reference without performing a restore preview and compatibility check.
+
+- --package-directory <PACKAGE_DIRECTORY>
+
+The directory where to restore the packages. The default package restore location is %userprofile%\.nuget\packages on Windows and ~/.nuget/packages on macOS and Linux. For more information, see Managing the global packages, cache, and temp folders in NuGet.
+
+- --prerelease
+
+Allows prerelease packages to be installed. Available since .NET Core 5 SDK
+
+- -s|--source <SOURCE>
+
+The URI of the NuGet package source to use during the restore operation.
+
+- -v|--version <VERSION>
```

Comment:
Can you specify version for a glob of packages as well?

Reply:
We will need to lock down the update strategies first I think. In general, I don't think so unless people have a compelling reason to do this!

Reply:
I remember having a conversation with @nkolev92 about how some packages need to have their version kept in sync, particularly in ASP.NET Core world. I believe a lot of those packages also share prefixes like these:

![image](https://user-images.githubusercontent.com/15097183/167951986-4f6d8b45-c0b6-46f5-8050-457be2fa4e2c.png)

Azure packages are also versioned similarly:

![image](https://user-images.githubusercontent.com/15097183/167952287-74caa4c1-1153-4a0a-a6b8-07343a0bc73e.png)

Reply:
That should be supported with current spec right?

Reply:
Yes, unless explicitly disallowed. I just wanted to clarify the intention 🙂